### PR TITLE
[BottomAppBar] Make FAB visibility a switch

### PR DESCRIPTION
**Pull request by romoore**
The BottomAppBar example should use a switch for toggling visiblity instead of
a "toggle cell".

Pivotal: https://www.pivotaltracker.com/story/show/157298385

**Before**
simulator screen shot - iphone 4s - 2018-05-04 at 14 13 16

**After**
simulator screen shot - iphone 4s - 2018-05-04 at 14 10 44

simulator screen shot - iphone 4s - 2018-05-04 at 14 10 46

### CODE REVIEW

File: components/BottomAppBar/examples/supplemental/BottomAppBarTypicalUseSupplemental.m [hunk 1]
```diff
@@ -66,15 +70,29 @@ - (instancetype)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibB
...
     _listItems = listItems;
   }
   return self;
 }
 
+- (void)dealloc {
+  [self.fabVisibilitySwitch removeTarget:self
+                                  action:@selector(didTapFABVisibilitySwitch:)
+                        forControlEvents:UIControlEventAllEvents];
+}
```
Comment (mohammadcazig) [inline components/BottomAppBar/examples/supplemental/BottomAppBarTypicalUseSupplemental.m]: not needed.

File: components/BottomAppBar/examples/supplemental/BottomAppBarTypicalUseSupplemental.m [hunk 2]
```diff
@@ -66,15 +70,29 @@ - (instancetype)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibB
...
   [super viewDidLoad];
 
+  self.fabVisibilitySwitch = [[UISwitch alloc] init];
+  self.fabVisibilitySwitch.on = !self.bottomBarView.floatingButtonHidden;
+  [self.fabVisibilitySwitch addTarget:self
+                               action:@selector(didTapFABVisibilitySwitch:)
+                     forControlEvents:UIControlEventValueChanged];
+
+
+  [self.tableView registerClass:[UITableViewCell class] forCellReuseIdentifier:@"cell"];
```
Comment (mohammadcazig) [inline components/BottomAppBar/examples/supplemental/BottomAppBarTypicalUseSupplemental.m]: [nit] make @"cell" a constant.

File: components/BottomAppBar/examples/supplemental/BottomAppBarTypicalUseSupplemental.m [hunk 3]
```diff
@@ -124,8 +148,6 @@ - (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath
                                             animated:YES];
       break;
     case 5:
```
Comment (mohammadcazig) [inline components/BottomAppBar/examples/supplemental/BottomAppBarTypicalUseSupplemental.m:145]: [nit] Lets make this the same action as clicking on the switch, makes usage easier.

**Review by mohammadcazig — CHANGES_REQUESTED** [review]

File: components/BottomAppBar/examples/supplemental/BottomAppBarTypicalUseSupplemental.m (hunk 1 above)
Comment (romoore) [reply components/BottomAppBar/examples/supplemental/BottomAppBarTypicalUseSupplemental.m]: Done

File: components/BottomAppBar/examples/supplemental/BottomAppBarTypicalUseSupplemental.m (hunk 2 above)
Comment (romoore) [reply components/BottomAppBar/examples/supplemental/BottomAppBarTypicalUseSupplemental.m]: Done

File: components/BottomAppBar/examples/supplemental/BottomAppBarTypicalUseSupplemental.m (hunk 3 above)
Comment (romoore) [reply components/BottomAppBar/examples/supplemental/BottomAppBarTypicalUseSupplemental.m:145]: Done

**Review by mohammadcazig — APPROVED** [review]
LGTM

**Pull request merged**